client/source/js/modules/model/cost-coverage-ctrl.js

```diff
@@ -385,19 +387,18 @@ define(['./module', 'underscore'], function (module, _) {
       var model = getPlotModel();
       model.coparams = $scope.coParams[graphIndex];
       model.effect =  effectNames[graphIndex];
-      if ( hasSomeElements(model.coparams) ){
-        var message = 'Cost-coverage curve plotting options should be either empty, or all present!';
+      if ( !$scope.areValidParams(model.coparams) ){
         modalService.inform(
-          function (){ null },
+          function () {},
           'Okay',
-          message,
+          $scope.optionsErrorMessage,
           'Error!'
         );
         return;
       }
 
-      if (hasNoElements(model.ccparams)) delete model.ccparams;
-      if (hasNoElements(model.coparams)) delete model.coparams;
```

Comment:
The intention here was to remove ccparams and coparams if they haven't been set. if ccparams=[] then checking for invalid elements will not suffice: such an array has no invalid elements.